aws/rust-runtime/aws-sigv4/src/http_request/canonical_request.rs

```diff
@@ -225,7 +225,7 @@ impl<'a> CanonicalRequest<'a> {
         }
 
         let mut signed_headers = Vec::with_capacity(canonical_headers.len());
-        for (name, _) in &canonical_headers {
+        for name in canonical_headers.keys() {
```

Comment:
This change ensures that each key appears only once in the `SignedHeaders` section.